CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -77,6 +77,10 @@ public static function create(&$params, $trxnEntityTable = NULL) {
       $entityFinancialTrxnParams['entity_table'] = $trxnEntityTable['entity_table'];
       $entityFinancialTrxnParams['entity_id'] = $trxnEntityTable['entity_id'];
     }
+    elseif (!empty($params['entity_id']) && !empty($params['entity_table'])) {
+      $entityFinancialTrxnParams['entity_table'] = $params['entity_table'];
+      $entityFinancialTrxnParams['entity_id'] = $params['entity_id'];
+    }
```

Comment:
What about changing this function to this -

```
  /**
   * Create a financial transaction object.
   *
   * @param array $params
   *   Values relating to the transaction..
   *
   * @param array $trxnEntityTable
   *   Deprecated: array of entity_id & entity_table. Put
   *   these in params instead.
   *
   * @return CRM_Core_BAO_FinancialTrxn
   */
  public static function create($params, $trxnEntityTable = array()) {
    $trxn = new CRM_Financial_DBO_FinancialTrxn();
    $trxn->copyValues($params);

    if (!CRM_Utils_Rule::currencyCode($trxn->currency)) {
      $trxn->currency = CRM_Core_Config::singleton()->defaultCurrency;
    }

    $trxn->save();

    // save to entity_financial_trxn table
    $entityFinancialTrxnParams
      = array_merge(array(
        'entity_table' => "civicrm_contribution",
        'financial_trxn_id' => $trxn->id,
        'amount' => $params['total_amount'],
      ), $trxnEntityTable, $params);

    if (empty($entityFinancialTrxnParams['entity_id'])) {
      $entityFinancialTrxnParams['entity_id'] = $params['contribution_id'];
    }

    $entityTrxn = new CRM_Financial_DAO_EntityFinancialTrxn();
    $entityTrxn->copyValues($entityFinancialTrxnParams);
    $entityTrxn->save();
    return $trxn;
  }
```

 I could only find two instances where $trxnEntityTable was passed in 

BAO_LineItem
```
          $trxnTable = array(
            'entity_id' => $newFinancialItem->id,
            'entity_table' => 'civicrm_financial_item',
          );
          $reverseTrxn = CRM_Core_BAO_FinancialTrxn::create($updateFinancialItemInfoValues['financialTrxn'], $trxnTable);
``` 

&& 
FinancialTrxn::createPremiumTrxn (function looks a bit broken to me but that is by the by)
```
      $trxnEntityTable['entity_table'] = 'civicrm_contribution';
      $trxnEntityTable['entity_id'] = $params['contributionId'];
      CRM_Core_BAO_FinancialTrxn::create($financialtrxn, $trxnEntityTable);
```